src/webui/api/synccontroller.cpp

```diff
@@ -399,11 +399,16 @@ void SyncController::maindataAction()
 
     data["torrents"] = torrents;
 
-    QVariantList categories;
-    for (auto i = session->categories().cbegin(); i != session->categories().cend(); ++i)
-        categories << i.key();
+    QVariantList categoriesList;
```

Comment:
You have chosen the wrong type for this task. You need to sync it as QVariantHash (with category names as keys and category info encoded as QVariantMap as values) to properly handle category changed/modified events. See "torrents" for example.

Reply:
In my testing the `QVariantList` seems to work great. Removing a category shows it in the `categories_removed` section of sync/maindata. Editing a category's save path results in the category being shown in both `categories` and `categories_removed`.

Reply:
>Editing a category's save path results in the category being shown in both `categories` and `categories_removed`.

It's wrong behavior. It's not so critical while category has only one property (save path). Besides "category edited" events are not so frequent. Although it contradicts the "sync" feature design. "Edited" event should send only changed data, not to be transformed into pair of "removed" + "added" events.

Reply:
I've switched it to a `QVariantHash` and now I'm seeing the intended behavior. Will push the changes.